component/azstorage/azstorage.go

```diff
@@ -326,15 +326,15 @@ func (az *AzStorage) ReadFile(options internal.ReadFileOptions) (data []byte, er
 	return az.storage.ReadBuffer(options.Handle.Path, 0, 0)
 }
 
-func (az *AzStorage) ReadInBuffer(options internal.ReadInBufferOptions) (len int, err error) {
+func (az *AzStorage) ReadInBuffer(options internal.ReadInBufferOptions) (length int, err error) {
 	//log.Trace("AzStorage::ReadInBuffer : Read %s from %d offset", h.Path, offset)
 
 	if options.Offset > options.Handle.Size {
 		return 0, syscall.ERANGE
 	}
 
-	var dataLen int64 = int64(cap(options.Data))
-	if options.Handle.Size < (options.Offset + int64(cap(options.Data))) {
+	var dataLen int64 = int64(len(options.Data))
```

Comment:
I guess in options we have length field, we shall rather rely on that instead of cap/len of slice.

Reply:
Ok rechecked we do not have any length fields in options here. Would it be good to add that as a fields in options and then rely on it.